server/repositories/orders.repository.js

```diff
@@ -0,0 +1,42 @@
+import { ObjectId } from 'mongodb'
+import { db } from './mongo.js'
+
+const ordersCollection = db.collection('orders')
+const cartCollection = db.collection('cart')
+ordersCollection.createIndex('orderId')
+
+export async function checkout(userId) {
+  const user = await ordersCollection.findOne({ userId })
+  if (!user) console.error('User not found')
+
+  if ((await cartCollection.findOne({ userId: userId })) === null) {
```

Comment:
Move this and the line 18 logic to a single variable and check the length of array instead of checking null here.

Reply:
use get cart from cart service.